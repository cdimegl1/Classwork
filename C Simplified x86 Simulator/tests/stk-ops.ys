main:
		 irmovq	    stack, %rsp
		 irmovq	    $0xdeadbeef, %rax
                 pushq      %rax
                 popq       %rbx
		 pushq	    %rsp
		 popq	    %rax
		 halt
		 
		 .pos	    0x200
stack:

