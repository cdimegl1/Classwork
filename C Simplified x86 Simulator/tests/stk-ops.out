   0:	0000	bubble
   1:	0000	bubble
   2:	0000	bubble
   3:	0000	bubble
   4:	0000	irmovq	$0x200, %rsp
   5:	000a	irmovq	$0xdeadbeef, %rax
   6:	0014	bubble
   7:	0014	bubble
   8:	0014	bubble
   9:	0014	pushq	%rax
  10:	0016	bubble
  11:	0016	bubble
  12:	0016	bubble
  13:	0016	popq	%rbx
  14:	0018	bubble
  15:	0018	bubble
  16:	0018	bubble
  17:	0018	pushq	%rsp
  18:	001a	bubble
  19:	001a	bubble
  20:	001a	bubble
  21:	001a	popq	%rax
  22:	001c	halt	
rax: 0000000000000200
rcx: 0000000000000000
rdx: 0000000000000000
rbx: 00000000deadbeef
rsp: 0000000000000200
rbp: 0000000000000000
rsi: 0000000000000000
rdi: 0000000000000000
 r8: 0000000000000000
 r9: 0000000000000000
r10: 0000000000000000
r11: 0000000000000000
r12: 0000000000000000
r13: 0000000000000000
r14: 0000000000000000
 pc: 000000000000001c
status: HLT
cc: Z=0 S=0 O=0
W[000001f8]: 0000000000000200
